$NetBSD: patch-aa,v 1.4 2009/03/25 23:01:48 he Exp $

NetBSD's powerpc assembler wants % in front of registers, and
the C compiler doesn't prepend _ to symbols it emits.

--- src/jit/ppc/asm.s.orig	2009-03-08 21:32:42.000000000 +0100
+++ src/jit/ppc/asm.s
@@ -1,25 +1,30 @@
 .text
     .align 12
+#ifdef __linux__
     .globl _Parrot_ppc_jit_restore_nonvolatile_registers
 _Parrot_ppc_jit_restore_nonvolatile_registers:
+#else
+    .globl Parrot_ppc_jit_restore_nonvolatile_registers
+Parrot_ppc_jit_restore_nonvolatile_registers:
+#endif
 
-    lfd     f14,-84(r1)
-    lfd     f15,-92(r1)
-    lfd     f16,-100(r1)
-    lfd     f17,-108(r1)
-    lfd     f18,-116(r1)
-    lfd     f19,-124(r1)
-    lfd     f20,-132(r1)
-    lfd     f21,-140(r1)
-    lfd     f22,-148(r1)
-    lfd     f23,-156(r1)
-    lfd     f24,-164(r1)
-    lfd     f25,-172(r1)
-    lfd     f26,-180(r1)
-    lfd     f27,-188(r1)
-    lfd     f28,-196(r1)
-    lfd     f29,-204(r1)
-    lfd     f30,-212(r1)
-    lfd     f31,-220(r1)
+    lfd     %f14,-84(%r1)
+    lfd     %f15,-92(%r1)
+    lfd     %f16,-100(%r1)
+    lfd     %f17,-108(%r1)
+    lfd     %f18,-116(%r1)
+    lfd     %f19,-124(%r1)
+    lfd     %f20,-132(%r1)
+    lfd     %f21,-140(%r1)
+    lfd     %f22,-148(%r1)
+    lfd     %f23,-156(%r1)
+    lfd     %f24,-164(%r1)
+    lfd     %f25,-172(%r1)
+    lfd     %f26,-180(%r1)
+    lfd     %f27,-188(%r1)
+    lfd     %f28,-196(%r1)
+    lfd     %f29,-204(%r1)
+    lfd     %f30,-212(%r1)
+    lfd     %f31,-220(%r1)
 
     blr
